### DISCUSSION TITLE
What do folks do for getting their preview deployments to talk to their preview backends hosted outside of vercel?

### DISCUSSION THREAD

**Discussion by sohcfst** [vercel/community#464] [Help]
If the url generated for a backend deployment is randomly generated hash appended to the branch name for example on a kubernetes cluster, how should a vercel preview deployment know about that specific service?

**Comment by samcx** [comment]
"how should a vercel preview deployment know about that specific service?" → Are you referring to the client-side of your Deployment reading the backend URL?

I would consider using a custom domain for your preview backends.

**Comment by wiiiimm** [comment]
How is your backend deployed on Kubernetes? If it is handled in CI/CD and you are creating a new frontend deployment for every backend deployment, you can consider deploying the frontend deployment using Vercel CLI as part of your CI/CD task. You can send in custom URLs returned by your backend deployment task as part of the CLI deployment parameter. You can find out more about this over at https://vercel.com/support/articles/using-vercel-cli-for-custom-workflows.

However, if backend deployments do not correspond to frontend deployments, then you can consider setting the URL env inside `.env` or hardcode it inside your code (if you are using Next.js, you _should_ do it inside `next.config.js`).
